--- a/cores/teensy4/usb_desc.h
+++ b/cores/teensy4/usb_desc.h
@@ -285,8 +285,8 @@
   #define DEVICE_PROTOCOL	0x01
   #define MANUFACTURER_NAME	{'T','e','e','n','s','y','d','u','i','n','o'}
   #define MANUFACTURER_NAME_LEN	11
-  #define PRODUCT_NAME		{'T','e','e','n','s','y','d','u','i','n','o',' ','S','e','r','i','a','l','+','K','e','y','b','o','a','r','d','+','M','o','u','s','e','+','J','o','y','s','t','i','c','k'}
-  #define PRODUCT_NAME_LEN	42
+  #define PRODUCT_NAME		{'T','e','e','n','s','y',' ','H','I','D',' ','M','u','l','t','i','p','l','e','x','e','r'}
+  #define PRODUCT_NAME_LEN	22
   #define EP0_SIZE		64
   #define NUM_ENDPOINTS		7
   #define NUM_INTERFACE		6
@@ -313,16 +313,21 @@
   #define MOUSE_ENDPOINT        6
   #define MOUSE_SIZE            8
   #define MOUSE_INTERVAL        2
-  #define JOYSTICK_INTERFACE    4	// Joystick
-  #define JOYSTICK_ENDPOINT     7
-  #define JOYSTICK_SIZE         12	//  12 = normal, 64 = extreme joystick
-  #define JOYSTICK_INTERVAL     2
+  #define RAWHID_USAGE_PAGE	0xFF60  // WebHID VIAL / VIA vendor usage page
+  #define RAWHID_USAGE		0x0061  // WebHID VIAL / VIA vendor usage
+  #define RAWHID_INTERFACE      4	// RawHID (VIAL / WebHID on EP7)
+  #define RAWHID_TX_ENDPOINT    7
+  #define RAWHID_TX_SIZE        32
+  #define RAWHID_TX_INTERVAL    1
+  #define RAWHID_RX_ENDPOINT    7
+  #define RAWHID_RX_SIZE        32
+  #define RAWHID_RX_INTERVAL    1
   #define ENDPOINT2_CONFIG	ENDPOINT_RECEIVE_UNUSED + ENDPOINT_TRANSMIT_INTERRUPT
   #define ENDPOINT3_CONFIG	ENDPOINT_RECEIVE_BULK + ENDPOINT_TRANSMIT_BULK
   #define ENDPOINT4_CONFIG      ENDPOINT_RECEIVE_UNUSED + ENDPOINT_TRANSMIT_INTERRUPT
   #define ENDPOINT5_CONFIG	ENDPOINT_RECEIVE_UNUSED + ENDPOINT_TRANSMIT_INTERRUPT
   #define ENDPOINT6_CONFIG      ENDPOINT_RECEIVE_UNUSED + ENDPOINT_TRANSMIT_INTERRUPT
-  #define ENDPOINT7_CONFIG	ENDPOINT_RECEIVE_UNUSED + ENDPOINT_TRANSMIT_INTERRUPT
+  #define ENDPOINT7_CONFIG	ENDPOINT_RECEIVE_INTERRUPT + ENDPOINT_TRANSMIT_INTERRUPT
 
 #elif defined(USB_TOUCHSCREEN)
   #define VENDOR_ID		0x16C0
